app/views/Settings.js

```diff
@@ -52,13 +54,42 @@ export const SettingsScreen = ({ navigation }) => {
     return () => {
       BackHandler.removeEventListener('hardwareBackPress', handleBackPress);
     };
-  }, [navigation]);
+  }, []);
+
+  //Initialyy when settings page is loaded , this method will check whether we have remove location permission from app settings.
+  // It will set status "Location Active" only if permission is granted.
+  const checkIsLocationEnable = () => {
+    BackgroundGeolocation.checkStatus(({ authorization }) => {
+      if (authorization === BackgroundGeolocation.AUTHORIZED) {
+        LocationServices.start();
+        setIsLogging(true);
+      } else if (authorization === BackgroundGeolocation.NOT_AUTHORIZED) {
+        LocationServices.stop();
+        setIsLogging(false);
+        BackgroundTaskServices.stop();
+      }
+    });
+  };
+
+  //  This method is used to track whether user has "Allow" or "Deny" location permission from system Dialog.
+  const SystemDialogPermission = async () => {
```

Comment:
methods should be `camelCase()`

It's unclear whether this is supposed to request permissions, or check the status of permissions?